### PR TITLE
Implement Cobra Command

### PR DESCRIPTION
**Pull request by thisisibrahimd**

### CODE REVIEW

**Comment by L1ghtman2k** [comment]
Were you able to test it on your local cluster with a release cut?

**Comment by thisisibrahimd** [comment]
> Were you able to test it on your local cluster with a release cut?

Not yet, will let you know my test result when I do.

**Comment by thisisibrahimd** [comment]
Perform testing and scoretrak works normally (from my understanding of what should fail and not)

Testing included:
Creating accounts, host groups, hosts, service groups, services.

**Comment by thisisibrahimd** [comment]
You can find the update helm chart here: https://github.com/ScoreTrak/helm-charts/tree/update-scoretrak-server

**Comment by L1ghtman2k** [comment]
> You can find the updated helm chart here: https://github.com/ScoreTrak/helm-charts/tree/update-scoretrak-server
Hey, i just added a few comments to the PR for update-scoretrak-server

**Comment by L1ghtman2k** [comment]
I saw you made bank optional in helm chart. Were you able to test that bank user gets created with proper (randomized) creds?

**Comment by thisisibrahimd** [comment]
> I saw you made bank optional in helm chart. Were you able to test that bank user gets created with proper (randomized) creds (since user creation is ran thru scoretrak-cli)?

Testing bank chart deployment soon

**Comment by thisisibrahimd** [comment]
Tested the bank charts with the new scoretrak and payment user was created successfully!

Noticed in pkg/docker/docker.go that a docker image was programmically being generated and wonder if it needs to change in any way.

**Comment by L1ghtman2k** [comment]
> Tested the bank charts with the new scoretrak and payment user was created successfully!
> 
> Noticed in pkg/platform/docker/docker.go that a docker image was programmically being generated and wonder if it needs to change in any way.

When we compile the binary, the version gets baked directly in the binary via `-ldflags` at compile time: https://github.com/ScoreTrak/ScoreTrak/blob/195a54692d1bf729802570236e9e837da2715636/Dockerfile#L10. We then use the above version to generate docker image programmatically, so client version is always the same version as master. I don't think we need to change it

**Comment by L1ghtman2k** [comment]
```
NAME                                       READY   STATUS             RESTARTS   AGE
internal-internal-7zmgm                    0/1     InvalidImageName   0          28s
internal-internal-ld9qx                    0/1     InvalidImageName   0          28s
internal-internal-pwb8k                    0/1     InvalidImageName   0          28s
```
```
    Image:         ghcr.io/scoretrak/scoretrak/scoretrak-server:No Version Provided
```

**Comment by thisisibrahimd** [comment]
> ```
> NAME                                       READY   STATUS             RESTARTS   AGE
> internal-internal-7zmgm                    0/1     InvalidImageName   0          28s
> internal-internal-ld9qx                    0/1     InvalidImageName   0          28s
> internal-internal-pwb8k                    0/1     InvalidImageName   0          28s
> ```
> 
> ```
>     Image:         ghcr.io/scoretrak/scoretrak/scoretrak-server:No Version Provided
> ```
> 
> I have a feeling it is caused by the fact that there is `go install` and `go build`, which is redundant since both produce binaries. And the `go install` step doesn't contain `ldflags`, hence the version is wrong

Simple typo from me, :face_palm:. Fixed versioning, cfg flag, and worker deployment issues. Tested deployment of workers with ping services 👍 .

Should be good to go!

**Comment by L1ghtman2k** [comment]
Sounds exciting, I will do a test right now, and if all good, we can merge this :D

File: pkg/queue/nsq/nsq.go [hunk 1]
```diff
@@ -56,15 +56,19 @@ func connectConsumer(consumer *nsq.Consumer, config queueing.Config) (err error)
...
 func validateNSQConfig(config queueing.Config) error {
 	if config.NSQ.ProducerNSQD == "" {
 		return ErrNSQDProducerAddressNotProvided
 	}
-	// Emulates exclusive-or
-	if (len(config.NSQ.NSQLookupd) != 0) != (len(config.NSQ.ConsumerNSQDPool) != 0) {
+	switch {
+	case len(config.NSQ.NSQLookupd) != 0 && len(config.NSQ.ConsumerNSQDPool) != 0:
 		return ErrProvidedBothNSQLookupdAndNSQD
+	case len(config.NSQ.NSQLookupd) == 0 && len(config.NSQ.ConsumerNSQDPool) == 0:
```
Comment (L1ghtman2k) [inline pkg/queue/nsq/nsq.go:69]: nice catch with our XOR bug 😂

Comment (thisisibrahimd) [reply pkg/queue/nsq/nsq.go:69]: No problem 🤣

**Comment by L1ghtman2k** [comment]
LGTM, feel free to merge (was able to test too)

**Review by L1ghtman2k — APPROVED** [review]

**Comment by thisisibrahimd** [comment]
> LGTM, feel free to merge (was able to test too)

Sounds good, going to squash and merge the pull request..

**Pull request merged**